router/core/router_config.go

```diff
@@ -0,0 +1,306 @@
+	postOriginHandlers              []TransportPostHandler
+	headerRules                     *config.HeaderRules
+	subgraphTransportOptions        *SubgraphTransportOptions
+	graphqlMetricsConfig            *GraphQLMetricsConfig
+	routerTrafficConfig             *config.RouterTrafficConfiguration
+	batchingConfig                  *BatchingConfig
+	fileUploadConfig                *config.FileUpload
+	accessController                *AccessController
+	retryOptions                    retrytransport.RetryOptions
+	redisClient                     rd.RDCloser
+	mcpServer                       *mcpserver.GraphQLSchemaServer
+	processStartTime                time.Time
+	developmentMode                 bool
+	healthcheck                     health.Checker
+	accessLogsConfig                *AccessLogsConfig
+	// If connecting to localhost inside Docker fails, fallback to the docker internal address for the host
+	localhostFallbackInsideDocker bool
+	tlsServerConfig               *tls.Config
+	tlsConfig                     *TlsConfig
+	telemetryAttributes           []config.CustomAttribute
+	tracePropagators              []propagation.TextMapPropagator
+	compositePropagator           propagation.TextMapPropagator
+	// Poller
+	configPoller                 configpoller.ConfigPoller
+	selfRegister                 selfregister.SelfRegister
+	registrationInfo             *nodev1.RegistrationInfo
+	securityConfiguration        config.SecurityConfiguration
+	customModules                []Module
+	engineExecutionConfiguration config.EngineExecutionConfiguration
+	// should be removed once the users have migrated to the new overrides config
+	overrideRoutingURLConfiguration config.OverrideRoutingURLConfiguration
+	// the new overrides config
+	overrides                  config.OverridesConfiguration
+	authorization              *config.AuthorizationConfiguration
+	rateLimit                  *config.RateLimitConfiguration
+	webSocketConfiguration     *config.WebSocketConfiguration
+	subgraphErrorPropagation   config.SubgraphErrorPropagationConfiguration
+	clientHeader               config.ClientHeader
+	cacheWarmup                *config.CacheWarmupConfiguration
+	multipartHeartbeatInterval time.Duration
+	hostName                   string
+	mcp                        config.MCPConfiguration
+	plugins                    config.PluginsConfiguration
+}
+
+// Usage returns an anonymized version of the config for usage tracking
+// The anonymized usage map is not containing any sensitive information
+// It's not a anonymized copy of the config, but a map of properties explaining how the Router is configured
+// The purpose is to get a high level understanding of which features are used
+// instead of how exactly the Router is configured
+func (c *Config) Usage() map[string]any {
+
+	type Exporter struct {
+		Name, Endpoint string
+	}
+
+	usage := make(map[string]any)
+
+	usage["cors"] = c.corsOptions != nil && c.corsOptions.Enabled
```

Comment:
Can you document the conventions, concept a bit? How would I extend it when I had to introduce a new feature?

Reply:
just add fields in the same way, no real concept